junit-jupiter-api/src/main/kotlin/org/junit/jupiter/api/Assertions.kt

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2015-2017 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v20.html
+ */
+@file:API(status = EXPERIMENTAL, since = "5.1")
+package org.junit.jupiter.api
+
+import org.apiguardian.api.API
+import org.apiguardian.api.API.Status.EXPERIMENTAL
+import org.junit.jupiter.api.function.Executable
+import java.util.function.Supplier
+import java.util.stream.Stream
+
+/**
+ * [Stream] of functions to be executed.
+ */
+internal typealias ExecutableStream = Stream<() -> Unit>
+internal fun ExecutableStream.convert() = map { Executable(it) }
```

Comment:
Can we make these private?

Reply:
Effectively they are. `internal` is different from java's "package private".
"Package private" in java means that if I (a consumer) create a file in a package called `org.junit.api` then I can access "package private" things. The same is not true in Kotlin for `internal`.
Internal is `internal` only to source code that is compiled together.

I can make them private if you wish though.

Reply:
So I was able to make them private.
I remember that in earlier versions of kotlin you couldn't call `private` things inside of `inline` functions. I'm not certain if that limitation still exists. But it doesn't seem to be a problem here.